lib/user_space/user_x86_64_atomic.h

```diff
@@ -114,13 +162,33 @@ static inline void m0_atomic64_sub(struct m0_atomic64 *a, int64_t num)
 static inline int64_t m0_atomic64_add_return(struct m0_atomic64 *a,
 						  int64_t delta)
 {
+#ifdef CONFIG_X86_64
 	long result;
 
 	result = delta;
 	asm volatile("lock xaddq %0, %1;"
 		     : "+r" (delta), "+m" (a->a_value)
 		     : : "memory");
 	return delta + result;
+#elif defined CONFIG_AARCH64
```

Comment:
Wait a minute: the name of this file suggests that it relates only (and only) to x86_64 platform! (I missed this point from the beginning, sorry.)

So it should be done like this: 1) similar to this file, introduce `user_aarch64_atomic.h` file and 2) add all your code there.

Reply:
It better to add one more tag to this file name like user_X86_64-Aarch64_atomic.h as it  is clearly contrast the  both platform code in one place  or  The file name should be  made more generic (user_platform_atomic.h)  to encompass all the platforms

Reply:
It's not better, and @nikitadanilov would not allow this. Please do as suggested.

Reply:
We can rename the file as  "user_platforms_atomic.h" that is more generic rather than naming the files explicitly with platform  name

Reply:
Fixed